CRM/Export/BAO/ExportProcessor.php

```diff
@@ -1542,21 +1520,22 @@ public function determineReturnProperties() {
     if ($this->getRequestedFields()) {
       $returnProperties = [];
       foreach ($this->getRequestedFields() as $key => $value) {
-        $fieldName = CRM_Utils_Array::value(1, $value);
-        if (!$fieldName || $this->isHouseholdMergeRelationshipTypeKey($fieldName)) {
+        $fieldName = $value['name'];
+        $locationName = CRM_Core_PseudoConstant::getName('CRM_Core_BAO_Address', 'location_type_id', $value['location_type_id']);
+        $relationshipTypeKey = !empty($value['relationship_type_id']) ? $value['relationship_type_id'] . '_' . $value['relationship_direction'] : NULL;
+        if (!$fieldName || $this->isHouseholdMergeRelationshipTypeKey($relationshipTypeKey)) {
           continue;
         }
 
-        if ($this->isRelationshipTypeKey($fieldName) && (!empty($value[2]) || !empty($value[4]))) {
-          $returnProperties[$fieldName] = $this->setRelationshipReturnProperties($value, $fieldName);
+        if ($this->isRelationshipTypeKey($relationshipTypeKey)) {
+          $returnProperties[$relationshipTypeKey] = $this->setRelationshipReturnProperties($value, $relationshipTypeKey);
         }
-        elseif (is_numeric(CRM_Utils_Array::value(2, $value))) {
-          $locationName = CRM_Core_PseudoConstant::getName('CRM_Core_BAO_Address', 'location_type_id', $value[2]);
-          if ($fieldName == 'phone') {
-            $returnProperties['location'][$locationName]['phone-' . CRM_Utils_Array::value(3, $value)] = 1;
+        elseif ($locationName) {
+          if ($fieldName === 'phone') {
+            $returnProperties['location'][$locationName]['phone-' . $value['phone_type_id'] ?? NULL] = 1;
           }
-          elseif ($fieldName == 'im') {
-            $returnProperties['location'][$locationName]['im-' . CRM_Utils_Array::value(3, $value)] = 1;
+          elseif ($fieldName === 'im') {
+            $returnProperties['location'][$locationName]['im-' . $value['im_provider_id'] ?? NULL] = 1;
```

Comment:
@eileenmcnaughton here is another Php7-only syntax. It seems to pass unit test so I guess Jenkins is on 7, but your blog says we support 5.6 for another 3 months: https://civicrm.org/blog/eileen/end-of-life-plans-for-5x-php-versions-planning-for-70-eol

Reply:
Nevermind, found the announcement.